src/main/java/nl/knaw/dans/catalog/core/UseCases.java

```diff
@@ -0,0 +1,182 @@
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class UseCases {
+    private final OcflObjectVersionRepository ocflObjectVersionRepository;
+    private final TarRepository tarRepository;
+    private final SearchIndex searchIndex;
+
+    private final TarEntityMapper tarEntityMapper = TarEntityMapper.INSTANCE;
+
+    public UseCases(OcflObjectVersionRepository ocflObjectVersionRepository, TarRepository tarRepository, SearchIndex searchIndex) {
+        this.ocflObjectVersionRepository = ocflObjectVersionRepository;
+        this.tarRepository = tarRepository;
+        this.searchIndex = searchIndex;
+    }
+
+    @UnitOfWork
+    public Collection<OcflObjectVersionEntity> findOcflObjectVersionsByBagId(String bagId) {
+        return ocflObjectVersionRepository.findAllByBagId(bagId);
+    }
+
+    @UnitOfWork
+    public Collection<OcflObjectVersionEntity> findOcflObjectVersionsBySwordToken(String swordToken) {
+        return ocflObjectVersionRepository.findAllBySwordToken(swordToken);
+    }
+
+    @UnitOfWork
+    public List<OcflObjectVersionEntity> findOcflObjectVersionsByNbn(String nbn) throws OcflObjectVersionNotFoundException {
+        var results = ocflObjectVersionRepository.findByNbn(nbn);
+
+        if (results.size() == 0) {
+            throw new OcflObjectVersionNotFoundException(
+                String.format("No OCFL object versions found for NBN %s", nbn)
+            );
+        }
+
+        return results;
+    }
+
+    @UnitOfWork
+    public Optional<OcflObjectVersionEntity> findOcflObjectVersionByBagIdAndVersion(String bagId, Integer versionNumber) {
+        return ocflObjectVersionRepository.findByBagIdAndVersion(bagId, versionNumber);
+    }
+
+    @UnitOfWork
+    public OcflObjectVersionEntity createOcflObjectVersion(OcflObjectVersionId id, OcflObjectVersionParameters parameters) throws OcflObjectVersionAlreadyExistsException {
+        var existingOcflObjectVersion = ocflObjectVersionRepository.findByBagIdAndVersion(id.getBagId(), id.getObjectVersion());
+
+        if (existingOcflObjectVersion.isPresent()) {
+            throw new OcflObjectVersionAlreadyExistsException(id.getBagId(), id.getObjectVersion());
+        }
+
+        var ocflObjectVersion = OcflObjectVersionEntity.builder()
+            .bagId(id.getBagId())
+            .objectVersion(id.getObjectVersion())
+            .swordToken(parameters.getSwordToken())
+            .nbn(parameters.getNbn())
+            .dataSupplier(parameters.getDataSupplier())
+            .dataversePid(parameters.getDataversePid())
+            .dataversePidVersion(parameters.getDataversePidVersion())
+            .otherId(parameters.getOtherId())
+            .otherIdVersion(parameters.getOtherIdVersion())
+            .ocflObjectPath(parameters.getOcflObjectPath())
+            .filePidToLocalPath(parameters.getFilePidToLocalPath())
+            .exportTimestamp(parameters.getExportTimestamp())
```

Comment:
I thought the convention for these method names was `with....`

Reply:
Oh, apparently generated with the Builder annotation.

Reply:
These are indeed openapi generated classes